features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/telemetry/MethodCalledTelemetry.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.telemetry
+
+import com.datadog.android.Datadog
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.sessionreplay.internal.recorder.telemetry.MetricBase.Companion.METRIC_TYPE
+
+internal class MethodCalledTelemetry(
+    private val callerClass: String,
+    private val logger: InternalLogger,
+    private val startTime: Long = System.nanoTime(),
+    private val internalSdkCore: InternalSdkCore? = Datadog.getInstance() as? InternalSdkCore
+) : MetricBase {
+    internal fun stopMethodCalled(isSuccessful: Boolean) {
```

Comment:
> [!CAUTION]
> Instead of making this method specific to this class and internal, you should make it generic in the `MetricBase` interfaces (e.g.: `MetricBase.stopMetric(isSuccessful)`). 
> Otherwise you're creating a tight coupling and bad practice as shown above: callers that get a MetricBase instance need to cast if `as? MethodCalledTelemetry` to be able to stop it. This means that any change to what instance is returned when calling `startMetric` will impact a lot of calling sites.